src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewGuidCommand.cs

```diff
@@ -49,7 +49,7 @@ protected override void ProcessRecord()
             }
             else
             {
-                guid = ParameterSetName is "Empty" ? Guid.Empty : Guid.NewGuid();
+                guid = Empty.ToBool() ? Guid.Empty : Guid.NewGuid();
```

Comment:
```suggestion
                guid = Empty.IsPresent ? Guid.Empty : Guid.NewGuid();
```

Reply:
this seems to be the more common pattern

Reply:
The use of `.ToBool()` instead of simply checking for presence is _the entire point_ of this PR, actually. The caller could pass in `-Empty:$false`, in which case it would be _present_, but the caller's intent is to _not_ get the empty GUID.

Reply:
Actually, I agree we shouldn't use `IsPresent`. It may be implemented identically, but this appears to be a design error.

Reply:
```suggestion
                guid = Empty ? Guid.Empty : Guid.NewGuid();
```

[The `SwitchParameter` type implicitly converts to Boolean](https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-7.5#:~:text=The%20SwitchParameter%20type%20implicitly%20converts%20to%20Boolean)

Reply:
> Actually, I agree we shouldn't use `IsPresent`. It may be implemented identically, but this appears to be a design error.

I've opened an issue: https://github.com/PowerShell/PowerShell/issues/26149

Reply:
If it is accepted, then this PR can be rebased to use `.IsSpecified`. If it is not accepted, then I posit that it is essentially not relevant whether this PR uses `.IsPresent` or `.ToBool()` -- but that `.IsPresent` makes less sense since the bug being fixed in this instance is precisely related to the fact that a switch might be present with the explicit value $false.

Reply:
>  _no_ places that use the implicit conversion

I find 379 references to the implicit operator, for example:

https://github.com/PowerShell/PowerShell/blob/be0542b1939de6a85ebf3e0e5ffd3c7c1429d0a8/src/System.Management.Automation/FormatAndOutput/out-console/OutConsole.cs#L194-L199

Reply:
That's really weird. To test, I _deleted_ the operator from the source code and ran `Start-PSBuild`, and it built with no errors. 🤔

Reply:
I didn't see errors at first. But I ran `Start-PSBuild -Clean` and now get error CS0029.

Reply:
Ahh, fair enough.